src/saturn_engine/worker_manager/config/declarative.py

```diff
@@ -1,55 +1,95 @@
 import dataclasses
 import os
+from collections import defaultdict
+from typing import DefaultDict
 from typing import Optional
 
 import yaml
 
 from .declarative_inventory import Inventory
+from .declarative_job_definition import JobDefinition
+from .declarative_topic_item import TopicItem
+from .static_definitions import StaticDefinitions
 
 
 @dataclasses.dataclass
-class StaticDefinitions:
-    inventories: list[Inventory]
+class UncompiledObject:
+    api_version: str
+    kind: str
+    data: dict
 
 
-def load_definitions_from_str(definitions: str) -> StaticDefinitions:
-    inventories: list[Inventory] = []
+def _load_uncompiled_objects_from_str(definitions: str) -> list[UncompiledObject]:
+    uncompiled_objects: list[UncompiledObject] = []
 
     for yaml_object in yaml.load_all(definitions, yaml.SafeLoader):
 
         if not yaml_object:
             continue
 
-        apiVersion: Optional[str] = yaml_object.get("apiVersion")
-        if not apiVersion:
+        api_version: Optional[str] = yaml_object.get("apiVersion")
+        if not api_version:
             raise Exception("Missing apiVersion")
-        elif apiVersion != "saturn.github.io/v1alpha1":
+        elif api_version != "saturn.github.io/v1alpha1":
             raise Exception(
-                f"apiVersion was {apiVersion},"
+                f"apiVersion was {api_version},"
                 "we only support saturn.github.io/v1alpha1"
             )
 
         object_kind: Optional[str] = yaml_object.get("kind")
+
         if not object_kind:
             raise Exception("All objects should have a kind")
-        elif object_kind == "SaturnInventory":
-            inventories.append(Inventory.schema().load(yaml_object))
-        else:
-            raise Exception(f"Unknown kind {object_kind}")
 
-    return StaticDefinitions(
-        inventories=inventories,
-    )
+        uncompiled_objects.append(
+            UncompiledObject(
+                api_version=api_version,
+                kind=object_kind,
+                data=yaml_object,
+            )
+        )
+
+    return uncompiled_objects
+
+
+def _compile_objects(uncompiled_objects: list[UncompiledObject]) -> StaticDefinitions:
+    objects_by_kind: DefaultDict[str, list[UncompiledObject]] = defaultdict(list)
+    for uncompiled_object in uncompiled_objects:
+        objects_by_kind[uncompiled_object.kind].append(uncompiled_object)
+
+    definitions: StaticDefinitions = StaticDefinitions()
+
+    for uncompiled_inventory in objects_by_kind.pop("SaturnInventory", list()):
+        inventory: Inventory = Inventory.schema().load(uncompiled_inventory.data)
+        definitions.inventories[inventory.metadata.name] = inventory.to_core_object()
+
+    for uncompiled_topic in objects_by_kind.pop("SaturnTopic", list()):
+        topic_item: TopicItem = Inventory.schema().load(uncompiled_topic.data)
```

Comment:
`Inventory.schema()`?

Reply:
duh. Fixed. And I fixed the tests so that we would have caught it.